
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../../function/labeltransfer/">
      
      
        <link rel="next" href="../../../License/">
      
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.5.3, mkdocs-material-9.5.15">
    
    
      
        <title>plot - pycyto</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.7e359304.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../assets/_mkdocstrings.css">
    
      <link rel="stylesheet" href="../../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#pycyto.plot" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="pycyto" class="md-header__button md-logo" aria-label="pycyto" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8h5Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            pycyto
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              plot
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3 3.19.09m3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95 2.06.05m-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31Z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="teal" data-md-color-accent="indigo"  aria-label="Switch to dark"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0-7 2.39 3.42C13.65 5.15 12.84 5 12 5c-.84 0-1.65.15-2.39.42L12 2M3.34 7l4.16-.35A7.2 7.2 0 0 0 5.94 8.5c-.44.74-.69 1.5-.83 2.29L3.34 7m.02 10 1.76-3.77a7.131 7.131 0 0 0 2.38 4.14L3.36 17M20.65 7l-1.77 3.79a7.023 7.023 0 0 0-2.38-4.15l4.15.36m-.01 10-4.14.36c.59-.51 1.12-1.14 1.54-1.86.42-.73.69-1.5.83-2.29L20.64 17M12 22l-2.41-3.44c.74.27 1.55.44 2.41.44.82 0 1.63-.17 2.37-.44L12 22Z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../../.." class="md-tabs__link">
        
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../../../Installation/" class="md-tabs__link">
        
  
    
  
  Installation

      </a>
    </li>
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../../Tutorial/Analyzing%20Zebrafish%2010hpf%20Embryo%20Stereo-seq%20Data%20with%20pycyto/" class="md-tabs__link">
          
  
  Tutorial

        </a>
      </li>
    
  

      
        
  
  
    
  
  
    
    
      
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../../function/format/" class="md-tabs__link">
          
  
  API Reference

        </a>
      </li>
    
  

    
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../../../License/" class="md-tabs__link">
        
  
    
  
  License

      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="pycyto" class="md-nav__button md-logo" aria-label="pycyto" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8h5Z"/></svg>

    </a>
    pycyto
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Installation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Installation
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Tutorial
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Tutorial
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../Tutorial/Analyzing%20Zebrafish%2010hpf%20Embryo%20Stereo-seq%20Data%20with%20pycyto/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Analyzing Zebrafish 10hpf Embryo Stereo seq Data with pycyto
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
      
        
        
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" checked>
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="">
            
  
  <span class="md-ellipsis">
    API Reference
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            API Reference
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_1" >
        
          
          <label class="md-nav__link" for="__nav_4_1" id="__nav_4_1_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    function
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4_1">
            <span class="md-nav__icon md-icon"></span>
            function
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/format/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Format
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/qc/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Qc
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/insert_tiff/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Insert tiff
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/filtercells/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Filtercells
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/filtergenes/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Filtergenes
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/lognormal/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Lognormal
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/vst_highly_variable_genes/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Vst highly variable genes
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/svg/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Svg
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/scale/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Scale
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/pca/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Pca
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/tsne/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Tsne
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/findneighbors/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Findneighbors
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/leiden/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Leiden
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/Spatial_cluster/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Spatial cluster
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/findmarkers/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Findmarkers
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/metacell/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Metacell
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/generate_membership_anndata/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Generate membership anndata
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/falsepositive/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Falsepositive
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../function/labeltransfer/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Labeltransfer
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    plot
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    plot
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot" class="md-nav__link">
    <span class="md-ellipsis">
      plot
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.metacell_purity" class="md-nav__link">
    <span class="md-ellipsis">
      metacell_purity
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.metacell_size" class="md-nav__link">
    <span class="md-ellipsis">
      metacell_size
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.pca_variance_ratio" class="md-nav__link">
    <span class="md-ellipsis">
      pca_variance_ratio
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.plot_cellcount_between_ref_query" class="md-nav__link">
    <span class="md-ellipsis">
      plot_cellcount_between_ref_query
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.plot_labeltransfer_score_distribution" class="md-nav__link">
    <span class="md-ellipsis">
      plot_labeltransfer_score_distribution
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.plot_labeltransfer_score_violin" class="md-nav__link">
    <span class="md-ellipsis">
      plot_labeltransfer_score_violin
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.plot_marker_dotplot" class="md-nav__link">
    <span class="md-ellipsis">
      plot_marker_dotplot
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.plot_marker_heatmap" class="md-nav__link">
    <span class="md-ellipsis">
      plot_marker_heatmap
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.plot_moran" class="md-nav__link">
    <span class="md-ellipsis">
      plot_moran
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.plot_spatial" class="md-nav__link">
    <span class="md-ellipsis">
      plot_spatial
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.plot_tsne" class="md-nav__link">
    <span class="md-ellipsis">
      plot_tsne
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.spatial_bigcell_plot" class="md-nav__link">
    <span class="md-ellipsis">
      spatial_bigcell_plot
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.variable_plot" class="md-nav__link">
    <span class="md-ellipsis">
      variable_plot
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.violin" class="md-nav__link">
    <span class="md-ellipsis">
      violin
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../../License/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    License
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot" class="md-nav__link">
    <span class="md-ellipsis">
      plot
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.metacell_purity" class="md-nav__link">
    <span class="md-ellipsis">
      metacell_purity
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.metacell_size" class="md-nav__link">
    <span class="md-ellipsis">
      metacell_size
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.pca_variance_ratio" class="md-nav__link">
    <span class="md-ellipsis">
      pca_variance_ratio
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.plot_cellcount_between_ref_query" class="md-nav__link">
    <span class="md-ellipsis">
      plot_cellcount_between_ref_query
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.plot_labeltransfer_score_distribution" class="md-nav__link">
    <span class="md-ellipsis">
      plot_labeltransfer_score_distribution
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.plot_labeltransfer_score_violin" class="md-nav__link">
    <span class="md-ellipsis">
      plot_labeltransfer_score_violin
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.plot_marker_dotplot" class="md-nav__link">
    <span class="md-ellipsis">
      plot_marker_dotplot
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.plot_marker_heatmap" class="md-nav__link">
    <span class="md-ellipsis">
      plot_marker_heatmap
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.plot_moran" class="md-nav__link">
    <span class="md-ellipsis">
      plot_moran
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.plot_spatial" class="md-nav__link">
    <span class="md-ellipsis">
      plot_spatial
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.plot_tsne" class="md-nav__link">
    <span class="md-ellipsis">
      plot_tsne
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.spatial_bigcell_plot" class="md-nav__link">
    <span class="md-ellipsis">
      spatial_bigcell_plot
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.variable_plot" class="md-nav__link">
    <span class="md-ellipsis">
      variable_plot
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycyto.plot.violin" class="md-nav__link">
    <span class="md-ellipsis">
      violin
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  <h1>plot</h1>

<div class="doc doc-object doc-module">



<a id="pycyto.plot"></a>
  <div class="doc doc-contents first">

  

  <div class="doc doc-children">










<div class="doc doc-object doc-function">



<h2 id="pycyto.plot.metacell_purity" class="doc doc-heading">
          <code class="highlight language-python"><span class="n">metacell_purity</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;skyblue&#39;</span><span class="p">,</span> <span class="n">save</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">360</span><span class="p">,</span> <span class="n">celltype</span><span class="o">=</span><span class="s1">&#39;leiden&#39;</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="mi">6</span><span class="p">))</span></code>

</h2>


  <div class="doc doc-contents ">
  
      <p>Plot the distribution of metacell purity.</p>
<p>This function calculates the purity of each metacell based on the predominant cell type and plots
the distribution of purities as a histogram along with a kernel density estimate (KDE) curve.</p>



  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>adata</code></td>
          <td>
                <code><span title="pycyto.labeltransfer.anndata.AnnData">AnnData</span></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Annotated data object containing metacell information.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>bins</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Number of bins for the histogram. Defaults to 20.</p>
            </div>
          </td>
          <td>
                <code>20</code>
          </td>
        </tr>
        <tr>
          <td><code>color</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Color of the histogram bars. Defaults to 'skyblue'.</p>
            </div>
          </td>
          <td>
                <code>&#39;skyblue&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>save</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>File path to save the plot. If None, the plot is not saved. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>dpi</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Dots per inch for the saved figure. Defaults to 360.</p>
            </div>
          </td>
          <td>
                <code>360</code>
          </td>
        </tr>
        <tr>
          <td><code>celltype</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Column name representing the cell type information in the AnnData object. Defaults to 'leiden'.</p>
            </div>
          </td>
          <td>
                <code>&#39;leiden&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>figsize</code></td>
          <td>
                <code>tuple</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Size of the figure in inches (width, height). Defaults to (8, 6).</p>
            </div>
          </td>
          <td>
                <code>(8, 6)</code>
          </td>
        </tr>
    </tbody>
  </table>

  </div>

</div>


<div class="doc doc-object doc-function">



<h2 id="pycyto.plot.metacell_size" class="doc doc-heading">
          <code class="highlight language-python"><span class="n">metacell_size</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;skyblue&#39;</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">save</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">360</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="mi">6</span><span class="p">))</span></code>

</h2>


  <div class="doc doc-contents ">
  
      <p>Plot the distribution of metacell sizes.</p>
<p>This function plots a histogram of metacell sizes along with a kernel density estimate (KDE) curve.</p>



  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>adata</code></td>
          <td>
                <code><span title="pycyto.labeltransfer.anndata.AnnData">AnnData</span></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Annotated data object containing metacell information.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>color</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Color of the histogram bars. Defaults to 'skyblue'.</p>
            </div>
          </td>
          <td>
                <code>&#39;skyblue&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>bins</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Number of bins for the histogram. Defaults to 20.</p>
            </div>
          </td>
          <td>
                <code>20</code>
          </td>
        </tr>
        <tr>
          <td><code>save</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>File path to save the plot. If None, the plot is not saved. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>dpi</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Dots per inch for the saved figure. Defaults to 360.</p>
            </div>
          </td>
          <td>
                <code>360</code>
          </td>
        </tr>
        <tr>
          <td><code>figsize</code></td>
          <td>
                <code>tuple</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Size of the figure in inches (width, height). Defaults to (8, 6).</p>
            </div>
          </td>
          <td>
                <code>(8, 6)</code>
          </td>
        </tr>
    </tbody>
  </table>

  </div>

</div>


<div class="doc doc-object doc-function">



<h2 id="pycyto.plot.pca_variance_ratio" class="doc doc-heading">
          <code class="highlight language-python"><span class="n">pca_variance_ratio</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;brown&#39;</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;o&#39;</span><span class="p">,</span> <span class="n">markersize</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="p">,</span> <span class="mi">4</span><span class="p">),</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">360</span><span class="p">,</span> <span class="n">xlabel</span><span class="o">=</span><span class="s1">&#39;Number of PCs&#39;</span><span class="p">,</span> <span class="n">ylabel</span><span class="o">=</span><span class="s1">&#39;Variance ratio&#39;</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">save</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">log</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">npc</span><span class="o">=</span><span class="mi">20</span><span class="p">)</span></code>

</h2>


  <div class="doc doc-contents ">
  
      <p>Plot the variance ratio of principal components obtained from PCA.</p>



  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>adata</code></td>
          <td>
                <code><span title="pycyto.labeltransfer.anndata.AnnData">AnnData</span></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>An AnnData object containing PCA results.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>color</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Color of the plot markers. Defaults to 'brown'.</p>
            </div>
          </td>
          <td>
                <code>&#39;brown&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>marker</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Marker style for the plot. Defaults to 'o'.</p>
            </div>
          </td>
          <td>
                <code>&#39;o&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>markersize</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Size of the plot markers. Defaults to 4.</p>
            </div>
          </td>
          <td>
                <code>4</code>
          </td>
        </tr>
        <tr>
          <td><code>figsize</code></td>
          <td>
                <code>tuple</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Figure size in inches. Defaults to (5, 4).</p>
            </div>
          </td>
          <td>
                <code>(5, 4)</code>
          </td>
        </tr>
        <tr>
          <td><code>dpi</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Dots per inch for the figure resolution. Defaults to 360.</p>
            </div>
          </td>
          <td>
                <code>360</code>
          </td>
        </tr>
        <tr>
          <td><code>xlabel</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Label for the x-axis. Defaults to 'Number of PCs'.</p>
            </div>
          </td>
          <td>
                <code>&#39;Number of PCs&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>ylabel</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Label for the y-axis. Defaults to 'Variance ratio'.</p>
            </div>
          </td>
          <td>
                <code>&#39;Variance ratio&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>title</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Title for the plot. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>save</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>File path to save the plot. If None, the plot is not saved. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>log</code></td>
          <td>
                <code>bool</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Whether to plot the variance ratio in log scale. Defaults to False.</p>
            </div>
          </td>
          <td>
                <code>False</code>
          </td>
        </tr>
        <tr>
          <td><code>npc</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Number of principal components to include in the plot. Defaults to 20.</p>
            </div>
          </td>
          <td>
                <code>20</code>
          </td>
        </tr>
    </tbody>
  </table>

  </div>

</div>


<div class="doc doc-object doc-function">



<h2 id="pycyto.plot.plot_cellcount_between_ref_query" class="doc doc-heading">
          <code class="highlight language-python"><span class="n">plot_cellcount_between_ref_query</span><span class="p">(</span><span class="n">adata_query</span><span class="p">,</span> <span class="n">adata_ref</span><span class="p">,</span> <span class="n">key1</span><span class="o">=</span><span class="s1">&#39;labeltransfer_celltype&#39;</span><span class="p">,</span> <span class="n">key2</span><span class="o">=</span><span class="s1">&#39;celltype&#39;</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> <span class="n">xrotation</span><span class="o">=</span><span class="mi">45</span><span class="p">,</span> <span class="n">xha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">,</span> <span class="n">save</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">title1</span><span class="o">=</span><span class="s1">&#39;scRNA-seq&#39;</span><span class="p">,</span> <span class="n">title2</span><span class="o">=</span><span class="s1">&#39;Stereo-seq&#39;</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">360</span><span class="p">)</span></code>

</h2>


  <div class="doc doc-contents ">
  
      <p>Plot cell count comparison between reference and query datasets.</p>
<p>This function plots a bar chart comparing the cell counts of different cell types between a reference dataset
and a query dataset.</p>



  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>adata_query</code></td>
          <td>
                <code><span title="pycyto.labeltransfer.anndata.AnnData">AnnData</span></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>An AnnData object representing the query dataset.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>adata_ref</code></td>
          <td>
                <code><span title="pycyto.labeltransfer.anndata.AnnData">AnnData</span></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>An AnnData object representing the reference dataset.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>key1</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Key in adata_query.obs representing the cell type information. Defaults to 'labeltransfer_celltype'.</p>
            </div>
          </td>
          <td>
                <code>&#39;labeltransfer_celltype&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>key2</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Key in adata_ref.obs representing the cell type information. Defaults to 'celltype'.</p>
            </div>
          </td>
          <td>
                <code>&#39;celltype&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>figsize</code></td>
          <td>
                <code>tuple</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Size of the figure in inches (width, height). Defaults to (12, 5).</p>
            </div>
          </td>
          <td>
                <code>(12, 5)</code>
          </td>
        </tr>
        <tr>
          <td><code>xrotation</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Rotation angle for x-axis tick labels. Defaults to 45.</p>
            </div>
          </td>
          <td>
                <code>45</code>
          </td>
        </tr>
        <tr>
          <td><code>xha</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Horizontal alignment of x-axis tick labels. Defaults to 'right'.</p>
            </div>
          </td>
          <td>
                <code>&#39;right&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>save</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>File path to save the plot. If None, the plot is not saved. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>title1</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Title for the query dataset plot. Defaults to 'scRNA-seq'.</p>
            </div>
          </td>
          <td>
                <code>&#39;scRNA-seq&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>title2</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Title for the reference dataset plot. Defaults to 'Stereo-seq'.</p>
            </div>
          </td>
          <td>
                <code>&#39;Stereo-seq&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>dpi</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Dots per inch for the saved figure. Defaults to 360.</p>
            </div>
          </td>
          <td>
                <code>360</code>
          </td>
        </tr>
    </tbody>
  </table>

  </div>

</div>


<div class="doc doc-object doc-function">



<h2 id="pycyto.plot.plot_labeltransfer_score_distribution" class="doc doc-heading">
          <code class="highlight language-python"><span class="n">plot_labeltransfer_score_distribution</span><span class="p">(</span><span class="n">adata_transfer</span><span class="p">,</span> <span class="n">bar_color</span><span class="o">=</span><span class="s1">&#39;skyblue&#39;</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="mi">6</span><span class="p">),</span> <span class="n">save</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">xrotation</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">360</span><span class="p">)</span></code>

</h2>


  <div class="doc doc-contents ">
  
      <p>Plot the distribution of label transfer maximum scores.</p>
<p>This function plots the distribution of maximum scores obtained during the label transfer process.</p>



  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>adata_transfer</code></td>
          <td>
                <code><span title="pycyto.labeltransfer.anndata.AnnData">AnnData</span></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>An AnnData object containing the label transfer results.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>bar_color</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Color of the bars in the histogram. Defaults to 'skyblue'.</p>
            </div>
          </td>
          <td>
                <code>&#39;skyblue&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>figsize</code></td>
          <td>
                <code>tuple</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Size of the figure in inches (width, height). Defaults to (8, 6).</p>
            </div>
          </td>
          <td>
                <code>(8, 6)</code>
          </td>
        </tr>
        <tr>
          <td><code>save</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>File path to save the plot. If None, the plot is not saved. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>xrotation</code></td>
          <td>
                <code>float</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Rotation angle for x-axis tick labels. Defaults to 0.</p>
            </div>
          </td>
          <td>
                <code>0</code>
          </td>
        </tr>
        <tr>
          <td><code>dpi</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Dots per inch for the figure resolution. Defaults to 360.</p>
            </div>
          </td>
          <td>
                <code>360</code>
          </td>
        </tr>
    </tbody>
  </table>

  </div>

</div>


<div class="doc doc-object doc-function">



<h2 id="pycyto.plot.plot_labeltransfer_score_violin" class="doc doc-heading">
          <code class="highlight language-python"><span class="n">plot_labeltransfer_score_violin</span><span class="p">(</span><span class="n">adata_transfer</span><span class="p">,</span> <span class="n">groupby</span><span class="o">=</span><span class="s1">&#39;id&#39;</span><span class="p">,</span> <span class="n">inner</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">save</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> <span class="n">xrotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">xha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;blue&#39;</span><span class="p">)</span></code>

</h2>


  <div class="doc doc-contents ">
  
      <p>Plot a violin plot of label transfer maximum scores.</p>
<p>This function plots a violin plot of maximum scores obtained during the label transfer process grouped by a specified category.</p>



  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>adata_transfer</code></td>
          <td>
                <code><span title="pycyto.labeltransfer.anndata.AnnData">AnnData</span></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>An AnnData object containing the label transfer results.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>groupby</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Key in adata_transfer.obs specifying the category by which to group the scores. Defaults to 'id'.</p>
            </div>
          </td>
          <td>
                <code>&#39;id&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>inner</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Type of data representation inside the violins. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>save</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>File path to save the plot. If None, the plot is not saved. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>figsize</code></td>
          <td>
                <code>tuple</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Size of the figure in inches (width, height). Defaults to (8, 5).</p>
            </div>
          </td>
          <td>
                <code>(8, 5)</code>
          </td>
        </tr>
        <tr>
          <td><code>xrotation</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Rotation angle for x-axis tick labels. Defaults to 90.</p>
            </div>
          </td>
          <td>
                <code>90</code>
          </td>
        </tr>
        <tr>
          <td><code>xha</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Horizontal alignment of x-axis tick labels. Defaults to 'right'.</p>
            </div>
          </td>
          <td>
                <code>&#39;right&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>color</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Color of the violins. Defaults to 'blue'.</p>
            </div>
          </td>
          <td>
                <code>&#39;blue&#39;</code>
          </td>
        </tr>
    </tbody>
  </table>

  </div>

</div>


<div class="doc doc-object doc-function">



<h2 id="pycyto.plot.plot_marker_dotplot" class="doc doc-heading">
          <code class="highlight language-python"><span class="n">plot_marker_dotplot</span><span class="p">(</span><span class="n">cdata</span><span class="p">,</span> <span class="n">genes</span><span class="o">=</span><span class="p">[],</span> <span class="n">groupby</span><span class="o">=</span><span class="s1">&#39;leiden&#39;</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">8</span><span class="p">),</span> <span class="n">vmin</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">vmax</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;viridis&#39;</span><span class="p">,</span> <span class="n">rotation</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">save</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">360</span><span class="p">)</span></code>

</h2>


  <div class="doc doc-contents ">
  
      <p>Generate a marker dot plot to visualize gene expression across cell clusters.</p>
<p>This function generates a marker dot plot to visualize the expression of marker genes across different cell
clusters. Each dot in the plot represents a cell cluster, and the color and size of the dot indicate the average
expression level and percentage of cells expressing the marker gene, respectively.</p>



  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>cdata</code></td>
          <td>
                <code><span title="pycyto.labeltransfer.anndata.AnnData">AnnData</span></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>An AnnData object containing the data.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>genes</code></td>
          <td>
                <code>list</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>List of marker genes to include in the dot plot.</p>
            </div>
          </td>
          <td>
                <code>[]</code>
          </td>
        </tr>
        <tr>
          <td><code>groupby</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Name of the category column in cdata.obs to group cells by. Defaults to 'leiden'.</p>
            </div>
          </td>
          <td>
                <code>&#39;leiden&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>figsize</code></td>
          <td>
                <code>tuple</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Size of the figure in inches (width, height). Defaults to (10, 8).</p>
            </div>
          </td>
          <td>
                <code>(10, 8)</code>
          </td>
        </tr>
        <tr>
          <td><code>vmin</code></td>
          <td>
                <code>float</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Minimum value for color normalization. Defaults to 0.</p>
            </div>
          </td>
          <td>
                <code>0</code>
          </td>
        </tr>
        <tr>
          <td><code>vmax</code></td>
          <td>
                <code>float</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Maximum value for color normalization. Defaults to 3.</p>
            </div>
          </td>
          <td>
                <code>3</code>
          </td>
        </tr>
        <tr>
          <td><code>cmap</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Colormap name for coloring the dots. Defaults to 'viridis'.</p>
            </div>
          </td>
          <td>
                <code>&#39;viridis&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>rotation</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Rotation angle for cluster labels on the x-axis. Defaults to 0.</p>
            </div>
          </td>
          <td>
                <code>0</code>
          </td>
        </tr>
        <tr>
          <td><code>save</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>File path to save the plot. If None, the plot is not saved. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>dpi</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Dots per inch for the figure resolution. Defaults to 360.</p>
            </div>
          </td>
          <td>
                <code>360</code>
          </td>
        </tr>
    </tbody>
  </table>

  </div>

</div>


<div class="doc doc-object doc-function">



<h2 id="pycyto.plot.plot_marker_heatmap" class="doc doc-heading">
          <code class="highlight language-python"><span class="n">plot_marker_heatmap</span><span class="p">(</span><span class="n">cdata</span><span class="p">,</span> <span class="n">use_raw</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">genes</span><span class="o">=</span><span class="p">[],</span> <span class="n">groupby</span><span class="o">=</span><span class="s1">&#39;leiden&#39;</span><span class="p">,</span> <span class="n">z_score</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">10</span><span class="p">),</span> <span class="n">vmin</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">vmax</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">save</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="s1">&#39;viridis&#39;</span><span class="p">,</span> <span class="n">standard_scale</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">rotation</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">360</span><span class="p">)</span></code>

</h2>


  <div class="doc doc-contents ">
  
      <p>Generate a marker heatmap plot for visualizing gene expression across cell clusters.</p>
<p>This function generates a marker heatmap plot to visualize the expression of marker genes across different cell
clusters. Each row in the heatmap represents a marker gene, and each column represents a cell cluster. The color
intensity of each cell in the heatmap indicates the expression level of the corresponding marker gene in the
respective cell cluster.</p>



  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>cdata</code></td>
          <td>
                <code><span title="pycyto.labeltransfer.anndata.AnnData">AnnData</span></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>An AnnData object containing the data.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>use_raw</code></td>
          <td>
                <code>bool</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Whether to use raw.X instead of .X for gene expression. Defaults to True.</p>
            </div>
          </td>
          <td>
                <code>True</code>
          </td>
        </tr>
        <tr>
          <td><code>genes</code></td>
          <td>
                <code>list</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>List of marker genes to include in the heatmap. Defaults to an empty list.</p>
            </div>
          </td>
          <td>
                <code>[]</code>
          </td>
        </tr>
        <tr>
          <td><code>groupby</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Name of the category column in cdata.obs to group cells by. Defaults to 'leiden'.</p>
            </div>
          </td>
          <td>
                <code>&#39;leiden&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>z_score</code></td>
          <td>
                <code>int or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Whether to z-score normalize the expression values. If None, no normalization is applied.
                   Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>figsize</code></td>
          <td>
                <code>tuple</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Size of the figure in inches (width, height). Defaults to (10, 10).</p>
            </div>
          </td>
          <td>
                <code>(10, 10)</code>
          </td>
        </tr>
        <tr>
          <td><code>vmin</code></td>
          <td>
                <code>float or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Minimum value for color normalization. If None, the minimum value in the data is used.
                  Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>vmax</code></td>
          <td>
                <code>float or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Maximum value for color normalization. If None, the maximum value in the data is used.
                  Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>save</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>File path to save the plot. If None, the plot is not saved. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>cmap</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Colormap name for coloring the heatmap. Defaults to 'viridis'.</p>
            </div>
          </td>
          <td>
                <code>&#39;viridis&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>standard_scale</code></td>
          <td>
                <code>int or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Whether to standard scale the expression values along the rows or columns. If None,
                          no scaling is applied. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>rotation</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Rotation angle for cluster labels on the x-axis. Defaults to 0.</p>
            </div>
          </td>
          <td>
                <code>0</code>
          </td>
        </tr>
        <tr>
          <td><code>dpi</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Dots per inch for the figure resolution. Defaults to 360.</p>
            </div>
          </td>
          <td>
                <code>360</code>
          </td>
        </tr>
    </tbody>
  </table>



  <p><strong>Returns:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Type</th>
        <th>Description</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td>
                <code><span title="seaborn.matrix.ClusterGrid">ClusterGrid</span></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>A ClusterGrid object representing the marker heatmap plot.</p>
            </div>
          </td>
        </tr>
    </tbody>
  </table>

  </div>

</div>


<div class="doc doc-object doc-function">



<h2 id="pycyto.plot.plot_moran" class="doc doc-heading">
          <code class="highlight language-python"><span class="n">plot_moran</span><span class="p">(</span><span class="n">moran1</span><span class="p">,</span> <span class="n">moran2</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="p">,</span> <span class="mi">5</span><span class="p">),</span> <span class="n">s</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">anno</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">save</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">360</span><span class="p">)</span></code>

</h2>


  <div class="doc doc-contents ">
  
      <p>Plot Moran scatter plot between two datasets.</p>
<p>This function plots a Moran scatter plot between two datasets. It visualizes the Moran scores of common genes
shared between the two datasets.</p>



  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>moran1</code></td>
          <td>
                <code><span title="pandas.Series">Series</span></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Moran scores for the first dataset.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>moran2</code></td>
          <td>
                <code><span title="pandas.Series">Series</span></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Moran scores for the second dataset.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>figsize</code></td>
          <td>
                <code>tuple</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Size of the figure in inches (width, height). Defaults to (5, 5).</p>
            </div>
          </td>
          <td>
                <code>(5, 5)</code>
          </td>
        </tr>
        <tr>
          <td><code>s</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Marker size. Defaults to 2.</p>
            </div>
          </td>
          <td>
                <code>2</code>
          </td>
        </tr>
        <tr>
          <td><code>anno</code></td>
          <td>
                <code>list or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>List of gene names to annotate on the plot. If None, no annotations are added. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>save</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>File path to save the plot. If None, the plot is not saved. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>dpi</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Dots per inch for the saved figure. Defaults to 360.</p>
            </div>
          </td>
          <td>
                <code>360</code>
          </td>
        </tr>
    </tbody>
  </table>

  </div>

</div>


<div class="doc doc-object doc-function">



<h2 id="pycyto.plot.plot_spatial" class="doc doc-heading">
          <code class="highlight language-python"><span class="n">plot_spatial</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">by</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">savefig</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">360</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">6</span><span class="p">),</span> <span class="n">tiff</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">coord</span><span class="o">=</span><span class="s1">&#39;spatial&#39;</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">categorical</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span></code>

</h2>


  <div class="doc doc-contents ">
  
      <p>Plot spatial information from Stereo-seq data.</p>
<p>This function visualizes spatial information from Stereo-seq data. It can display cell
coordinates on an image, with the option to color cells based on a specified category or gene expression level.</p>



  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>adata</code></td>
          <td>
                <code><span title="pycyto.labeltransfer.anndata.AnnData">AnnData</span></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>An AnnData object containing the spatial data.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>by</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Name of the category column in adata.obs to group cells by, or name of the gene for coloring
              cells based on gene expression. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>s</code></td>
          <td>
                <code>float</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Size of the markers representing cells. Defaults to 5.</p>
            </div>
          </td>
          <td>
                <code>5</code>
          </td>
        </tr>
        <tr>
          <td><code>cmap</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Colormap for coloring cells. Defaults to None, which uses the 'viridis' colormap.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>savefig</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>File path to save the plot. If None, the plot is not saved. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>dpi</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Dots per inch for the figure resolution. Defaults to 360.</p>
            </div>
          </td>
          <td>
                <code>360</code>
          </td>
        </tr>
        <tr>
          <td><code>figsize</code></td>
          <td>
                <code>tuple</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Size of the figure in inches (width, height). Defaults to (6, 6).</p>
            </div>
          </td>
          <td>
                <code>(6, 6)</code>
          </td>
        </tr>
        <tr>
          <td><code>tiff</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Name of the TIFF file in adata.uns['spatial'] to overlay on the plot. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>coord</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Key in adata.obsm containing the spatial coordinates. Defaults to 'spatial'.</p>
            </div>
          </td>
          <td>
                <code>&#39;spatial&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>title</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Title for the plot. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>legend</code></td>
          <td>
                <code>bool</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Whether to display the legend. Defaults to False.</p>
            </div>
          </td>
          <td>
                <code>False</code>
          </td>
        </tr>
        <tr>
          <td><code>categorical</code></td>
          <td>
                <code>bool</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>If True, color cells based on the specified category. If False, color cells based on gene
                expression. Defaults to False.</p>
            </div>
          </td>
          <td>
                <code>False</code>
          </td>
        </tr>
    </tbody>
  </table>

  </div>

</div>


<div class="doc doc-object doc-function">



<h2 id="pycyto.plot.plot_tsne" class="doc doc-heading">
          <code class="highlight language-python"><span class="n">plot_tsne</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">by</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">cmap</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">savefig</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">360</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">6</span><span class="p">))</span></code>

</h2>


  <div class="doc doc-contents ">
  
      <p>Generate a t-SNE plot for visualizing high-dimensional data.</p>
<p>This function generates a t-SNE plot to visualize high-dimensional data in two dimensions. Data points are represented
as scatter points on the plot, with optional coloring by a specified category or gene expression level.</p>



  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>adata</code></td>
          <td>
                <code><span title="pycyto.labeltransfer.anndata.AnnData">AnnData</span></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>An AnnData object containing the data.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>by</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Name of the category column in adata.obs or a gene name in adata.var_names to color the plot
              by. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>s</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Marker size for data points. Defaults to 5.</p>
            </div>
          </td>
          <td>
                <code>5</code>
          </td>
        </tr>
        <tr>
          <td><code>cmap</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Colormap name for coloring the plot. If None, 'viridis' colormap is used. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>savefig</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>File path to save the plot. If None, the plot is not saved. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>dpi</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Dots per inch for the figure resolution. Defaults to 360.</p>
            </div>
          </td>
          <td>
                <code>360</code>
          </td>
        </tr>
        <tr>
          <td><code>figsize</code></td>
          <td>
                <code>tuple</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Size of the figure in inches (width, height). Defaults to (6, 6).</p>
            </div>
          </td>
          <td>
                <code>(6, 6)</code>
          </td>
        </tr>
    </tbody>
  </table>

  </div>

</div>


<div class="doc doc-object doc-function">



<h2 id="pycyto.plot.spatial_bigcell_plot" class="doc doc-heading">
          <code class="highlight language-python"><span class="n">spatial_bigcell_plot</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">ncol</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mi">15</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">7</span><span class="p">,</span> <span class="mi">7</span><span class="p">),</span> <span class="n">line_color</span><span class="o">=</span><span class="s1">&#39;red&#39;</span><span class="p">,</span> <span class="n">linewidth</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">xlabel</span><span class="o">=</span><span class="s1">&#39;X&#39;</span><span class="p">,</span> <span class="n">ylabel</span><span class="o">=</span><span class="s1">&#39;Y&#39;</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">save</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">360</span><span class="p">,</span> <span class="n">spatial_bigcell_key</span><span class="o">=</span><span class="s1">&#39;membership&#39;</span><span class="p">,</span> <span class="n">color_key</span><span class="o">=</span><span class="s1">&#39;color&#39;</span><span class="p">,</span> <span class="n">color_key_obs</span><span class="o">=</span><span class="s1">&#39;leiden&#39;</span><span class="p">)</span></code>

</h2>


  <div class="doc doc-contents ">
  
      <p>Plot the spatial distribution of cells with highlighted spatial metacells.</p>
<p>This function visualizes the spatial distribution of cells and highlights spatial metacells using convex hulls.</p>



  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>adata</code></td>
          <td>
                <code><span title="pycyto.labeltransfer.anndata.AnnData">AnnData</span></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>An AnnData object containing spatial coordinates and membership information.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>ncol</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Number of columns for the legend. Defaults to 1.</p>
            </div>
          </td>
          <td>
                <code>1</code>
          </td>
        </tr>
        <tr>
          <td><code>s</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Marker size for individual cells. Defaults to 15.</p>
            </div>
          </td>
          <td>
                <code>15</code>
          </td>
        </tr>
        <tr>
          <td><code>figsize</code></td>
          <td>
                <code>tuple</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Figure size in inches. Defaults to (7, 7).</p>
            </div>
          </td>
          <td>
                <code>(7, 7)</code>
          </td>
        </tr>
        <tr>
          <td><code>line_color</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Color of the convex hull lines. Defaults to 'red'.</p>
            </div>
          </td>
          <td>
                <code>&#39;red&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>linewidth</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Line width of the convex hull lines. Defaults to 1.</p>
            </div>
          </td>
          <td>
                <code>1</code>
          </td>
        </tr>
        <tr>
          <td><code>xlabel</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Label for the x-axis. Defaults to 'X'.</p>
            </div>
          </td>
          <td>
                <code>&#39;X&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>ylabel</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Label for the y-axis. Defaults to 'Y'.</p>
            </div>
          </td>
          <td>
                <code>&#39;Y&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>title</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Title for the plot. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>save</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>File path to save the plot. If None, the plot is not saved. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>dpi</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Dots per inch for the figure resolution. Defaults to 360.</p>
            </div>
          </td>
          <td>
                <code>360</code>
          </td>
        </tr>
        <tr>
          <td><code>spatial_bigcell_key</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Key in adata.obs representing spatial big cell membership. Defaults to 'membership'.</p>
            </div>
          </td>
          <td>
                <code>&#39;membership&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>color_key</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Key in adata.obs representing the color of individual cells. Defaults to 'color'.</p>
            </div>
          </td>
          <td>
                <code>&#39;color&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>color_key_obs</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Key in adata.obs representing the color for the legend. Defaults to 'leiden'.</p>
            </div>
          </td>
          <td>
                <code>&#39;leiden&#39;</code>
          </td>
        </tr>
    </tbody>
  </table>

  </div>

</div>


<div class="doc doc-object doc-function">



<h2 id="pycyto.plot.variable_plot" class="doc doc-heading">
          <code class="highlight language-python"><span class="n">variable_plot</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">cols</span><span class="o">=</span><span class="p">(</span><span class="s1">&#39;black&#39;</span><span class="p">,</span> <span class="s1">&#39;red&#39;</span><span class="p">),</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">6</span><span class="p">),</span> <span class="n">save</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">log</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">s</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">annotate_genes</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">legend_loc</span><span class="o">=</span><span class="s1">&#39;best&#39;</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">360</span><span class="p">,</span> <span class="n">anno_fontsize</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span></code>

</h2>


  <div class="doc doc-contents ">
  
      <p>Generate a variable feature plot to visualize mean expression versus variance of genes.</p>
<p>This function generates a variable feature plot to visualize the relationship between mean expression and variance
of genes. Highly variable genes are highlighted in a different color.</p>



  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>adata</code></td>
          <td>
                <code><span title="pycyto.labeltransfer.anndata.AnnData">AnnData</span></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>An AnnData object containing the data.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>cols</code></td>
          <td>
                <code>tuple</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Colors to use for plotting non-highly variable genes and highly variable genes, respectively.
          Defaults to ('black', 'red').</p>
            </div>
          </td>
          <td>
                <code>(&#39;black&#39;, &#39;red&#39;)</code>
          </td>
        </tr>
        <tr>
          <td><code>figsize</code></td>
          <td>
                <code>tuple</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Size of the figure in inches (width, height). Defaults to (6, 6).</p>
            </div>
          </td>
          <td>
                <code>(6, 6)</code>
          </td>
        </tr>
        <tr>
          <td><code>save</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>File path to save the plot. If None, the plot is not saved. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>log</code></td>
          <td>
                <code>bool</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Whether to use logarithmic scale for axes. Defaults to False.</p>
            </div>
          </td>
          <td>
                <code>False</code>
          </td>
        </tr>
        <tr>
          <td><code>s</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Marker size. Defaults to 1.</p>
            </div>
          </td>
          <td>
                <code>1</code>
          </td>
        </tr>
        <tr>
          <td><code>annotate_genes</code></td>
          <td>
                <code>list or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>List of genes to annotate on the plot. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>legend_loc</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Location of the legend. Defaults to 'best'.</p>
            </div>
          </td>
          <td>
                <code>&#39;best&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>dpi</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Dots per inch for the figure resolution. Defaults to 360.</p>
            </div>
          </td>
          <td>
                <code>360</code>
          </td>
        </tr>
        <tr>
          <td><code>anno_fontsize</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Font size for gene annotations. Defaults to 10.</p>
            </div>
          </td>
          <td>
                <code>10</code>
          </td>
        </tr>
    </tbody>
  </table>

  </div>

</div>


<div class="doc doc-object doc-function">



<h2 id="pycyto.plot.violin" class="doc doc-heading">
          <code class="highlight language-python"><span class="n">violin</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">groupby</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">feature</span><span class="o">=</span><span class="s1">&#39;n_genes&#39;</span><span class="p">,</span> <span class="n">n_rows</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">fig_size</span><span class="o">=</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="mi">6</span><span class="p">),</span> <span class="n">save_path</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">360</span><span class="p">)</span></code>

</h2>


  <div class="doc doc-contents ">
  
      <p>Generate violin plots to visualize the distribution of a feature across groups.</p>
<p>This function generates violin plots to visualize the distribution of a specified feature across different groups
in the dataset.</p>



  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>adata</code></td>
          <td>
                <code><span title="pycyto.labeltransfer.anndata.AnnData">AnnData</span></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>An AnnData object containing the data.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>groupby</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Key in adata.obs to group the data. If None, a single violin plot for the entire
                    dataset will be generated. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>feature</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Feature to visualize. Defaults to 'n_genes'.</p>
            </div>
          </td>
          <td>
                <code>&#39;n_genes&#39;</code>
          </td>
        </tr>
        <tr>
          <td><code>n_rows</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Number of rows in the plot grid. Defaults to 1.</p>
            </div>
          </td>
          <td>
                <code>1</code>
          </td>
        </tr>
        <tr>
          <td><code>fig_size</code></td>
          <td>
                <code>tuple</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Size of the figure in inches (width, height). Defaults to (8, 6).</p>
            </div>
          </td>
          <td>
                <code>(8, 6)</code>
          </td>
        </tr>
        <tr>
          <td><code>save_path</code></td>
          <td>
                <code>str or None</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>File path to save the plot. If None, the plot is not saved. Defaults to None.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>dpi</code></td>
          <td>
                <code>int</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Dots per inch for the figure resolution. Defaults to 360.</p>
            </div>
          </td>
          <td>
                <code>360</code>
          </td>
        </tr>
    </tbody>
  </table>

  </div>

</div>



  </div>

  </div>

</div>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../../function/labeltransfer/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Labeltransfer">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Labeltransfer
              </div>
            </div>
          </a>
        
        
          
          <a href="../../../License/" class="md-footer__link md-footer__link--next" aria-label="Next: License">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                License
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../../..", "features": ["navigation.instant", "navigation.tabs", "navigation.tracking", "toc.follow", "navigation.top", "search.suggest", "navigation.footer"], "search": "../../../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.bd41221c.min.js"></script>
      
    
  </body>
</html>